src/components/PeoplePage.tsx

```diff
@@ -1,35 +1,83 @@
+import { getPeople } from '../api';
+import { Person } from '../types';
+import { getPreparedPeople } from '../helpers/GetPreparedPeople';
 
 export const PeoplePage = () => {
-  return (
-    <>
-      <h1 className="title">People Page</h1>
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
 
-      <div className="block">
-        <div className="columns is-desktop is-flex-direction-row-reverse">
-          <div className="column is-7-tablet is-narrow-desktop">
-            <PeopleFilters />
-          </div>
+  const [searchParams] = useSearchParams();
+  const sex = searchParams.get('sex') || '';
+  const query = searchParams.get('query') || '';
+  const sort = searchParams.get('sort') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+  const order = searchParams.get('order') || '';
+
+  useEffect(() => {
+    setIsLoading(true);
+    getPeople()
+      .then((loadingPeople) => setPeople(loadingPeople))
+      .catch(() => setIsError(true))
+      .finally(() => setIsLoading(false));
+  }, []);
 
-          <div className="column">
-            <div className="box table-container">
-              <Loader />
+  const peopleForRender = getPreparedPeople(people, {
+    sort,
+    query,
+    sex,
+    centuries,
+    order,
+  });
 
-              <p data-cy="peopleLoadingError">Something went wrong</p>
+  return (
+    <>
+      <h1 className="title">People Page</h1>
+      {isLoading
+        ? <Loader />
+        : (
+          <div className="block">
+            <div className="columns is-desktop is-flex-direction-row-reverse">
+              <div className="column is-7-tablet is-narrow-desktop">
+                <PeopleFilters />
+              </div>
 
-              <p data-cy="noPeopleMessage">
-                There are no people on the server
-              </p>
+              <div className="column">
+                <div className="box table-container">
+                  {isError && !isLoading && (
+                    <p data-cy="peopleLoadingError" className="has-text-danger">
+                      Something went wrong
+                    </p>
+                  )}
 
-              <p>There are no people matching the current search criteria</p>
+                  {!people.length && !isLoading && (
+                    <p data-cy="noPeopleMessage">
+                      There are no people on the server
+                    </p>
+                  )}
 
-              <PeopleTable />
+                  {!peopleForRender.length && !isLoading
+                    ? (
+                      <p>
+                        There are no people matching the current search criteria
+                      </p>
+                    ) : (
+                      people.length > 0 && !isLoading && (
+                        <PeopleTable people={peopleForRender} />
+                      )
+                    )}
```

Comment:
```suggestion
                  {!peopleForRender.length
                    ? (
                      <p>
                        There are no people matching the current search criteria
                      </p>
                    ) : (
                      <PeopleTable people={peopleForRender} />
                    )}
```
So many redundant conditions, pay attention